tiledb/sm/cpp_api/deleter.h

```diff
@@ -138,6 +139,13 @@ class Deleter {
     tiledb_consolidation_plan_free(&p);
   }
 
+  void operator()(tiledb_string_t* p) const {
+    capi_return_t result = tiledb_string_free(&p);
```

Comment:
Should use `tiledb_status_code` here to convert a return type to a status type. This is for future-proofing C API calls.